internal/zinc-core/src/main/scala/sbt/internal/inc/Lookup.scala

```diff
@@ -129,6 +129,8 @@ trait ExternalLookup extends ExternalHooks.Lookup {
     import scala.collection.JavaConverters._
     shouldDoIncrementalCompilation(changedClasses.iterator().asScala.toSet, previousAnalysis)
   }
+
+  def hasAnyMacro(a: Analysis): Boolean = a.apis.internal.values.exists(_.hasMacro)
```

Comment:
```suggestion
  /**
   * Used to override whether we should proceed with making an early output.
   * By default we do not make an early output in the presence of any macros
   * because macros require bytecode JARs in the classpath.
   * However, with careful micromanagement there are cases where we can safely
   * start downstream subproject compilations with just the pickle JARs.
   */
  def shouldDoEarlyOutput(analysis: CompileAnalysis): Boolean =
    analysis.asInstanceOf[Analysis].apis.internal.values.exists(_.hasMacro)
```

Reading the commit comment

> If you carefully micro-manage macro implementation scopes, you can avoid bailing here.

I'm not against the direction, but if this is the case, I think the name `hasAnyMacro` is misleading.